sqlmesh/dbt/loader.py

```diff
@@ -31,6 +31,36 @@
     from sqlmesh.core.context import GenericContext
 
 
+def get_default_dialect(
+    model_defaults_config: ModelDefaultsConfig, profile: Profile, profile_name: t.Optional[str]
+) -> ModelDefaultsConfig:
```

Comment:
I don't think this method is needed. Either you've specified a default dialect in sqlmesh, the model has a dialect via config (that will be set in basemodel, or you use the target. I believe all we need to do is set the model defaults in context and let the overrides of that happen later in the model logic.